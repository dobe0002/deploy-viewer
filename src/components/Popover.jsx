import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Popover, PopoverHeader, PopoverBody } from 'reactstrap';

const BS4Popover = ({ id, popOverTitle, body, hover }) => {
  const [popoverOpen, setPopoverOpen] = useState(false);

  const toggle = () => setPopoverOpen(!popoverOpen);

  return (
    <div>
      <Popover
        placement='bottom'
        isOpen={popoverOpen}
        target={id}
        toggle={toggle}
        trigger={hover ? 'hover focus' : 'click'}
      >
        <PopoverHeader>{popOverTitle}</PopoverHeader>
        <PopoverBody>{body}</PopoverBody>
      </Popover>
    </div>
  );
};

BS4Popover.propTypes = {
  id: PropTypes.string,
  popOverTitle: PropTypes.string,
  body: PropTypes.string,
  hover: PropTypes.bool
};
BS4Popover.defaultProps = {
  id: 'popoverbutton',
  popOverTitle: '',
  body: '',
  hover: true
};

export default BS4Popover;
