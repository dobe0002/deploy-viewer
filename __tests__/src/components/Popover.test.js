/* eslint-disable react/jsx-filename-extension */
import React from 'react';

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Popover from '../../../src/components/Popover';

configure({ adapter: new Adapter() });

const { axe, toHaveNoViolations } = require('jest-axe');

expect.extend(toHaveNoViolations);

describe('Popover component tests', () => {
  const sampleProps = {
    id: 'myPopOverId',
    popOverTitle: 'This is my popover title',
    body: 'This is my popover body'
  };
  test('Accessibility check', async () => {
    const div = document.createElement('div');
    div.id = sampleProps.id;
    document.body.appendChild(div);
    const wrapper = mount(
      <Popover
        id={sampleProps.id}
        popOverTitle={sampleProps.popOverTitle}
        body={sampleProps.body}
      />,
      { attachTo: div }
    );
    const results = await axe(wrapper.getDOMNode());
    // console.log('Axe violations', results.violations);
    expect(results).toHaveNoViolations();
    // console.log(wrapper.html());
  });

  test('Snapshot test', () => {
    const div = document.createElement('div');
    div.id = sampleProps.id;
    document.body.appendChild(div);
    const loading = mount(
      <Popover
        id={sampleProps.id}
        popOverTitle={sampleProps.popOverTitle}
        body={sampleProps.body}
      />,
      { attachTo: div }
    );
    expect(loading).toMatchSnapshot();
  });
});
