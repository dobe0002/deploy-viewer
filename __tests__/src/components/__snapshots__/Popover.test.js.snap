// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover component tests Snapshot test 1`] = `
<BS4Popover
  body="This is my popover body"
  hover={true}
  id="myPopOverId"
  popOverTitle="This is my popover title"
>
  <div>
    <Popover
      isOpen={false}
      placement="bottom"
      placementPrefix="bs-popover"
      target="myPopOverId"
      toggle={[Function]}
      trigger="hover focus"
    >
      <TooltipPopoverWrapper
        autohide={false}
        delay={
          Object {
            "hide": 0,
            "show": 0,
          }
        }
        fade={true}
        hideArrow={false}
        innerClassName="popover-inner"
        isOpen={false}
        placement="bottom"
        placementPrefix="bs-popover"
        popperClassName="popover show"
        target="myPopOverId"
        toggle={[Function]}
        trigger="hover focus"
      />
    </Popover>
  </div>
</BS4Popover>
`;
